crates/astria-bridge-contracts/src/lib.rs

```diff
@@ -454,32 +457,35 @@ where
                 .expect("must be set if this method is entered"),
         );
 
-        let memo = memo_to_json(&memos::v1::Ics20WithdrawalFromRollup {
+        let ics20_withdrawal_from_rollup = memos::v1::Ics20WithdrawalFromRollup {
             memo: event.memo.clone(),
             rollup_block_number,
             rollup_return_address: event.sender.encode_hex(),
             rollup_withdrawal_event_id,
-        })
-        .map_err(GetWithdrawalActionsError::encode_memo)?;
+        };
+
+        let memo = serde_json::to_string(&ics20_withdrawal_from_rollup)
+            .map_err(GetWithdrawalActionsError::serialize_memo)?;
 
         let amount = calculate_amount(&event, self.asset_withdrawal_divisor)
             .map_err(GetWithdrawalActionsError::calculate_withdrawal_amount)?;
 
-        let action = Ics20Withdrawal {
+        let action = Ics20Withdrawal::FromRollup(Box::new(Ics20WithdrawalWithBridgeAddress {
             denom,
             destination_chain_address: event.destination_chain_address,
             return_address: self.bridge_address,
             amount,
-            memo,
+            ics20_withdrawal_from_rollup,
```

Comment:
why do we have this `memos::v1::Ics20WithdrawalFromRollup` and the json-encoded memo? we can probably omit the json-encoded one here, and leave that to the sequencer to convert the `memos::v1::Ics20WithdrawalFromRollup` to json before putting it in the IBC packet

Reply:
Related to discussion below: https://github.com/astriaorg/astria/pull/1770#discussion_r1870169100